stk/calculators/energy/energy_calculators.py

```diff
@@ -377,3 +382,847 @@ def energy(self, mol, conformer=-1):
+            The path to the xTB executable.
+
+        gfn_version : :class:`str`
+            Parameterization of GFN to use in xTB.
+            For details:
+                https://xtb-docs.readthedocs.io/en/latest/basics.html
+
+        output_dir : :class:`str`, optional
+            The name of the directory into which files generated during
+            the optimization are written, if ``None`` then
+            :func:`uuid.uuid4` is used.
+
+        num_cores : :class:`int`, optional
+            The number of cores for xTB to use. Requires appropriate setup
+            of xTB by user.
+
+        etemp : :class:`int`, optional
+            Electronic temperature to use (in K). Defaults to 300K.
+
+        solvent : :class:`str`, optional
+            Solvent to use in GBSA implicit solvation method.
+            For details:
+                https://xtb-docs.readthedocs.io/en/latest/gbsa.html
+
+        solvent_grid : :class:`str`, optional
+            Grid level to use in SASA calculations for GBSA implicit solvent.
+            Options:
+                normal, tight, verytight, extreme
+            For details:
+                https://xtb-docs.readthedocs.io/en/latest/gbsa.html
+
+        charge : :class:`int`, optional
+            Formal molecular charge.
+
+        unpaired_electrons : :class:`int`, optional
+            Number of unpaired electrons.
+
+        use_cache : :class:`bool`, optional
+            If ``True`` :meth:`energy` will not run twice on the same
+            molecule and conformer.
+
+        mem_ulimit : :class: `bool`, optional
+            If ``True`` :meth:`energy` will be run without constraints on
+            the stacksize. If memory issues are encountered, this should be
+            ``True``, however this may raise issues on clusters.
+
+        """
+        self.xtb_path = xtb_path
+        self.gfn_version = gfn_version
+        self.output_dir = output_dir
+        self.num_cores = str(num_cores)
+        self.etemp = str(etemp)
+        self.solvent = solvent
+        if self.solvent is not None:
+            self.solvent = solvent.lower()
+            valid_xtb_solvent(
+                gfn_version=self.gfn_version,
+                solvent=self.solvent
+            )
+        self.solvent_grid = solvent_grid
+        self.charge = str(charge)
+        self.unpaired_electrons = str(unpaired_electrons)
+        self.mem_ulimit = mem_ulimit
+
+        self.total_energies = {}
+        self.homo_lumo_gaps = {}
+        self.fermi_levels = {}
+        self.homo_lumo_orbitals = {}
+        self.qonly_dipole_moments = {}
+        self.full_dipole_moments = {}
+        self.qonly_quadrupole_moments = {}
+        self.qdip_quadrupole_moments = {}
+        self.full_quadrupole_moments = {}
+        super().__init__(use_cache=use_cache)
+
+    def _get_properties(self, mol, conformer, output_file):
+        """
+        Extracts desired properties from GFN-xTB single point energy
```

Comment:
The first line of a docstring should be a sentence no longer than 71 characters.